---
bool isComplete
bool isStuck
bool isLeft
